### ISSUE TITLE
Inconsistent naming of variables in Discrete TabularCPD

### ISSUE THREAD

**Issue by daanknoope** [pgmpy/pgmpy#1062]
### Subject of the issue
When a discrete CPD is learned (e.g. using MLE), the group to which a variable belongs is denoted using parenthesis behind the variable (for example `A(0)` for the case in which A=0), but when the `copy` function is used, this changes into underscores (i.e. `A_0`). This is inconsistent and makes it hard to work with the tables later on.

### Your environment
* pgmpy version: 0.1.7
* Python version: 3.7.2
* Operating System: Ubuntu 18.04.2

### Steps to reproduce
1. Learn a discrete tabular cpd from data, using the example from the source code and print it:

```python
import pandas as pd
import numpy as np
from pgmpy.models import BayesianModel
model = BayesianModel([('A', 'B'), ('B', 'C'),('A', 'D'), ('D', 'C')])
values = pd.DataFrame(np.random.randint(low=0, high=2, size=(1000, 4)),columns=['A', 'B', 'C', 'D'])
model.fit(values)
print(model.get_cpds()[0])
```

2. Create a copy of a CPD and print it:
```python
print(model.get_cpds()[0].copy())
```

### Expected behaviour
The expected behaviour would be two completely equal tables, so:

```
+------+-------+
| A(0) | 0.479 |
+------+-------+
| A(1) | 0.521 |
+------+-------+
+------+-------+
| A(0) | 0.479 |
+------+-------+
| A(1) | 0.521 |
+------+-------+
```

### Actual behaviour
The parentheses are changed to underscores:

```
+------+-------+
| A(0) | 0.479 |
+------+-------+
| A(1) | 0.521 |
+------+-------+
+-----+-------+
| A_0 | 0.479 |
+-----+-------+
| A_1 | 0.521 |
+-----+-------+
```

**Comment by khalibartan** [comment]
@daanknoope The bug is pretty simple to fix. The `copy` method of `TabularCPD` is not using `state_names` while creating the CPD. If you want you can send a PR. Changes need to be done at [this  line](https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/factors/discrete/DiscreteFactor.py#L682)

**Comment by daanknoope** [comment]
Thanks, will fix it!

**Comment by ankurankan** [comment]
Fixed in #1063
